api/src/main/java/org/pmiops/workbench/featuredworkspace/FeaturedWorkspaceServiceImpl.java

```diff
@@ -49,23 +51,23 @@ public Optional<FeaturedWorkspaceCategory> getFeaturedCategory(DbWorkspace dbWor
 
   public List<WorkspaceResponse> getWorkspaceResponseByFeaturedCategory(
       FeaturedWorkspaceCategory featuredWorkspaceCategory) {
-    DbFeaturedCategory requestedDbCategory =
-        featuredWorkspaceMapper.toDbFeaturedCategory(featuredWorkspaceCategory);
 
-    return featuredWorkspaceDao.findDbFeaturedWorkspacesByCategory(requestedDbCategory).stream()
-        .map(
-            dbFeaturedCategory -> {
-              DbWorkspace dbWorkspace = dbFeaturedCategory.getWorkspace();
-              RawlsWorkspaceResponse rawlsWorkspaceResponse =
-                  fireCloudService.getWorkspace(
-                      dbWorkspace.getWorkspaceNamespace(), dbWorkspace.getFirecloudName());
-              return workspaceMapper.toApiWorkspaceResponse(
-                  workspaceMapper.toApiWorkspace(
-                      dbWorkspace,
-                      rawlsWorkspaceResponse.getWorkspace(),
-                      initialCreditsExpirationService),
-                  rawlsWorkspaceResponse.getAccessLevel());
-            })
-        .toList();
+    List<DbWorkspace> dbWorkspaces =
+        featuredWorkspaceDao
+            .findDbFeaturedWorkspacesByCategory(
+                featuredWorkspaceMapper.toDbFeaturedCategory(featuredWorkspaceCategory))
+            .stream()
+            .map(DbFeaturedWorkspace::getWorkspace)
+            .toList();
+
+    Map<String, RawlsWorkspaceListResponse> fcWorkspacesByUuid =
```

Comment:
build a list of Terra (FC) workspaces by first calling `fireCloudService.getWorkspaces()` as we do in other situations.  This returns all workspaces that the user has access to.

Previously, we called `getWorkspace()` on each individual workspace, and unexpectedly the user did not have access to some of these.